### DISCUSSION TITLE
Stored procedure grants and ownership

### DISCUSSION THREAD

**Discussion by marksabincarnivaluk** [littleK0i/SnowDDL#31] [Q&A] [answered]
Hi,
I can't figure out how to handle this under the role hierarchy model. I have a procedure that I run as owner, owner is a privileged role with access to other schemas and tables required to run the procedure. Under the SnowDDL model the procedure is owned by the owner_s_role which has only local schema permissions.

**Comment by littleK0i** [comment, marked as answer]
Please check `SCHEMA` parameters `owner_schema_read`, `owner_schema_write`: https://docs.snowddl.com/basic/yaml-configs/schema

It grants READ or WRITE access to other schemas to the OWNER role of the current schema. So `owner_s_role` will be able to perform actions on objects in other schemas.

It is useful not only for procedures, but also for views.
